### PR TITLE
standardize k8s version and common setup steps in templates

### PR DESCRIPTION
**Pull request by eljohnson92**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR does some cleanup of our templates in a few ways
* executes common steps such as turning swap off in a consistent way across flavors
* removes providerID specification since the CCM can handle setting this correctly when it is installed
* sets all flavors to use the KUBERNETES_VERSION env variable so the `clusterctl --kubernetes-version` flag will work as expected

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/clusterclass-kubeadm/kubeadmConfigTemplate.yaml [hunk 1]
```diff
@@ -28,7 +28,10 @@ spec:
...
           permissions: "0500"
       preKubeadmCommands:
-        - /kubeadm-pre-init.sh '{{ ds.meta_data.label }}' ${KUBERNETES_VERSION}
+        - /kubeadm-pre-init.sh ${KUBERNETES_VERSION}
+        - sed -i '/swap/d' /etc/fstab
+        - swapoff -a
+        - hostnamectl set-hostname '{{ ds.meta_data.label }}' && hostname -F /etc/hostname
       joinConfiguration:
         nodeRegistration:
           kubeletExtraArgs:
```
Comment (AshleyDumaine) [inline templates/flavors/clusterclass-kubeadm/kubeadmConfigTemplate.yaml:37]: should we also remove `provider-id` here?

File: templates/flavors/default/kubeadmControlPlane.yaml
Comment (AshleyDumaine) [inline templates/flavors/default/kubeadmControlPlane.yaml:1]: same on this file as https://github.com/linode/cluster-api-provider-linode/pull/259/files#r1569316324

File: templates/flavors/clusterclass-kubeadm/kubeadmConfigTemplate.yaml (hunk 1 above)
Comment (eljohnson92) [reply templates/flavors/clusterclass-kubeadm/kubeadmConfigTemplate.yaml:37]: good catch, missed this one

**Review by AshleyDumaine — APPROVED** [review]
LGTM, gave it a go on the RKE2 and default flavors and works fine 👍

**Pull request merged**